pkg/config/config.go

```diff
@@ -512,6 +512,10 @@ type MachineConfig struct {
 	Image string `toml:"image,omitempty"`
 	// Memory in MB a machine is created with.
 	Memory uint64 `toml:"memory,omitempty,omitzero"`
+	// Repository to download images from
+	RepoURI string `toml:"repo_uri,omitempty"`
```

Comment:
I still do not think RepoURI is needed. Image can already contain a full URL.

Reply:
I agree just spell this out in image, with the entire URL

Reply:
@rhatdan do you mean as a note in the comment & docs about the possible values it can have including the URL option? 

@Luap99 I do agree it is not likely someone would want to change RepoURI, since the place you point it has to be a github repo that follows the same branching, structure, and filename patterns as docker-brew-fedora. I'm happy to move it back to the constant approach.

Reply:
I am questioning what is the difference between an image and a repoURI.  If I want to change to a ubuntu image, do I have to change both fields, or just the image?

Reply:
Ah gotcha. Ok so it's not possible to statically reference a Fedora image in docker-brew-fedora since they dont preserve file names on every push after an image update. They drop the files and republish a new name which is fedora + major version + update timestamp. So we have to have two behavior patterns dynamic discovery based on accepted common names, and a static specification to an arbitrary static tarball location. 

The previously existing qemu implementation has a similar need for two strategies (coreos stream name + static URL). This was achieved by overloading the meaning of the Image value (image = testing -> go fetch from official fcos, everything else assume it's a url). The WSL PR just mirrors that approach, except instead of streams it uses major version number( image = 34 -> go pull it from the official fedora channel, image = something else, access directly)

For the Ubuntu use-case it would be just updating the Image to a static file location.  You would not need to touch RepoURI. The dynamic case is very specific to the Linux distribution delivery channels so I can't think of many use cases to point RepoURI elsewhere.

Reply:
Can we add somekind of glob or regex expression here?
I am just interested on how we would do this if someone wanted to specify a RHEL image or an Ubuntu image?

Reply:
Currently, you could do (although requires a fix to some minor bugs in main that are resoled in the wsl PR):
```
podman machine init --image-path https://github.com/tianon/docker-brew-ubuntu-core/blob/dist-amd64/impish/ubuntu-impish-oci-amd64-root.tar.gz?raw=true
```
You could also set this in Image under containers conf. (Although this will of course fail during booting, since in the qemu case there is no ignition, and in the wsl case no dnf.) 

As you can see their convention is very different. While the fedora repo uses releases for branches and arch for directories, ubuntu uses arch for branches and codenames for directories. Arch Linux is on gitlab and stuff there files into "packages": https://gitlab.archlinux.org/archlinux/archlinux-docker/-/packages

Since there is no standard registry for rootfs tarballs, I think the only reliable way for us to support new download streams (not statically specifying like above) is with a small amount of distro specific code like we have for fcos and fedora. We could potentially add a config item in the future to specify the image stream name (fedora, fcos, rhel, etc) if there is ever a need to support multiple on a single platform

Reply:
BTW I went ahead and dropped RepoURI from this PR since it seems we all agree its not necessary. I can add it back if though if anyone has second thoughts.